#! /usr/bin/env python

PACKAGE='anafi_autonomy'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

trajectory_enum = gen.enum([	gen.const("none", 	int_t, 	0, 	"No command"),
				gen.const("hover", 	int_t, 	1, 	"Hover at (0, 0, 1, yaw_d)"),
				gen.const("user", 	int_t, 	2, 	"Defined by user with (x_d, y_d, z_d, yaw_d)")
				], "Trajectory type")

#       	Name		Type      	Level	Description				Default 	Min	Max 	Values
gen.add(	"trajectory", 	int_t,		1,	"Trajectory type",			0,		0,	2,	edit_method=trajectory_enum)
gen.add(	"x_d",   	double_t,	2,    	"Desired x position",			0,		-10,	10)
gen.add(	"y_d",   	double_t,    	2,    	"Desired y position",			0,		-10,	10)
gen.add(	"z_d", 	double_t,    	2,    	"Desired z position", 			1,		0,	2)
gen.add(	"yaw_d", 	double_t,    	2,    	"Desired yaw orientation", 		0,		-180,	180)

exit(gen.generate(PACKAGE, "anafi_autonomy", "setTrajectory"))
